pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/ConcurrentMapPartitionUpsertMetadataManager.java

```diff
@@ -323,11 +347,13 @@ static class RecordLocation {
     private final IndexSegment _segment;
     private final int _docId;
     private final Comparable _comparisonValue;
+    private final boolean _isDeletedRecord;
```

Comment:
Ideally we don't want to add field here since this will add 1 byte per entry. Currently each `RecordLocation` is 4 bytes, and this will make it 5 bytes.

Reply:
From the queryableDocIds, we should be able to tell whether a primary key is deleted

Reply:
Good point! Removed this field from RecordLocation.